packages/flutter/lib/src/widgets/two_pane.dart

```diff
@@ -0,0 +1,482 @@
+  ///     [TextDirection.rtl]
+  ///
+  /// On a vertical layout, where [direction] is [Axis.vertical]:
+  ///
+  ///   * The first pane is at the top when [verticalDirection] is
+  ///     [VerticalDirection.down]
+  ///   * The first pane is at the bottom when [verticalDirection] is
+  ///     [VerticalDirection.up]
+  ///
+  /// If [panePriority] is [TwoPanePriority.start], this is the only pane
+  /// visible.
+  final Widget startPane;
+
+  /// The second pane.
+  ///
+  /// On a horizontal layout, where [direction] is [Axis.horizontal]:
+  ///
+  ///   * The second pane is on the right when [textDirection] is
+  ///     [TextDirection.ltr]
+  ///   * The second pane is on the left when [textDirection] is
+  ///     [TextDirection.rtl]
+  ///
+  /// On a vertical layout, where [direction] is [Axis.vertical]:
+  ///
+  ///   * The second pane is at the bottom when [verticalDirection] is
+  ///     [VerticalDirection.down]
+  ///   * The second pane is at the top when [verticalDirection] is
+  ///     [VerticalDirection.up]
+  ///
+  /// If [panePriority] is [TwoPanePriority.end], this is the only pane
+  /// visible.
+  final Widget endPane;
+
+  /// Proportion of the available space occupied by the first pane. The second
+  /// pane takes over the rest of the screen.
+  ///
+  /// A value of 0.5 will make the 2 panes equal.
+  ///
+  /// This property is ignored is the screen is split into sub-screens by a
+  /// [DisplayFeature], in which case each pane takes over one sub-screen.
+  final double paneProportion;
+
+  /// Same as [Flex.textDirection].
+  ///
+  /// If the resolved [direction] is [Axis.horizontal], [textDirection] defaults
+  /// to the ambient [Directionality]. If none is provided, then the widget
+  /// asserts during build in debug mode.
+  final TextDirection? textDirection;
+
+  /// Same as [Flex.verticalDirection].
+  ///
+  /// Defaults to [VerticalDirection.down].
+  final VerticalDirection verticalDirection;
+
+  /// Same as [Flex.direction].
+  ///
+  /// This property is ignored is the screen is split into sub-screens by a
+  /// [DisplayFeature], in which case the direction is:
+  ///
+  ///   * [Axis.horizontal] when the sub-screens are located left and right.
+  ///   * [Axis.vertical] when the sub-screens are located top and bottom.
+  ///
+  /// Defaults to [Axis.horizontal].
+  final Axis direction;
+
+  /// Whether to show only one pane and which one, or both.
+  ///
+  /// This property is ignored is the screen is split into sub-screens by a
+  /// [DisplayFeature], in which case each pane takes over one sub-screen.
+  ///
+  /// Defaults to [TwoPanePriority.both].
+  final TwoPanePriority panePriority;
+
+  /// The padding between TwoPane and the edges of the screen.
+  ///
+  /// Used to align TwoPane with the real position of display features. It
+  /// measures the distance between TwoPane and the ambient MediaQuery.
+  ///
+  /// Defaults to [EdgeInsets.zero].
+  final EdgeInsets inset;
```

Comment:
This should probably be `padding` to remain consistent with the framework

Reply:
Ok, it was padding but at the last minute I found it confusing and renamed it. Will change back 👍